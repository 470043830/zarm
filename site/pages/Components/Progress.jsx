import Markdown from '@site/components/Markdown';

export default class Page extends Markdown {
  document = () => {
    return {
      document: require('@/components/progress/demo.md'),
      className: 'progress-page',
    };
  }
}
