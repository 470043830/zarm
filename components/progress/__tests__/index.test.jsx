import React from 'react';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Progress from '../index';

describe('Progress', () => {
  it('renders correctly', () => {
    const wrapper = render(<Progress percent={10} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders has children correctly', () => {
    const wrapper = render(<Progress percent={10} weight="thin">foo</Progress>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders shape is circle correctly', () => {
    const wrapper = render(<Progress type="circle" shape="rect" percent={10}>foo</Progress>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders shape is semi-circle correctly', () => {
    const wrapper = render(<Progress type="semi-circle" percent={10}>foo</Progress>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders percent correctly', () => {
    const wrapper = mount(<Progress type="circle" percent={10}>foo</Progress>);
    wrapper.setProps({ percent: 50 });
    expect(wrapper.props().percent).toEqual(50);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('size=lg & type=line & theme=primary as expected', () => {
    const wrapper = render(<Progress type="line" percent={10}>percentText</Progress>);
    wrapper.setProps({ size: 'lg' });
    expect(wrapper.props().size).toEqual('lg');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
