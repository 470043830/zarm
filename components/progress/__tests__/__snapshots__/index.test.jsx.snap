// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress renders correctly 1`] = `
<div
  class="za-progress type-line"
>
  <div
    class="za-progress-track"
    style="height:8px;border-radius:8px"
  >
    <div
      class="za-progress-thumb theme-default"
      style="width:10%;border-radius:8px"
    />
  </div>
</div>
`;

exports[`Progress renders has children correctly 1`] = `
<div
  class="za-progress type-line"
>
  <div
    class="za-progress-track"
    style="height:8px;border-radius:8px"
  >
    <div
      class="za-progress-thumb theme-default"
      style="width:10%;border-radius:8px"
    />
  </div>
  <div
    class="za-progress-text"
  >
    foo
  </div>
</div>
`;

exports[`Progress renders shape is circle correctly 1`] = `
<div
  class="za-progress type-line"
>
  <div
    class="za-progress-track"
    style="height:8px;border-radius:0"
  >
    <div
      class="za-progress-thumb theme-default"
      style="width:10%;border-radius:0"
    />
  </div>
  <div
    class="za-progress-text"
  >
    foo
  </div>
</div>
`;
